packages/amplify-frontend-android/lib/amplify-config-helper.js

```diff
@@ -0,0 +1,167 @@
+function generateConfig(context, amplifyConfig) {
+  const metaData = context.amplify.getProjectMeta();
+  amplifyConfig = amplifyConfig || {
+    UserAgent: 'aws-amplify-cli/2.0',
+    Version: '1.0',
+  };
+  constructAnalytics(metaData, amplifyConfig);
+  constructApi(metaData, amplifyConfig);
+  constructPredictions(metaData, amplifyConfig);
+  constructStorage(metaData, amplifyConfig);
+
+  return amplifyConfig;
+}
+
+function constructAnalytics(metaData, amplifyConfig) {
+  const categoryName = 'analytics';
+  const pluginName = 'awsPinpointAnalyticsPlugin';
+  if (metaData[categoryName] && Object.keys(metaData[categoryName]).length > 0) {
+    const r = Object.keys(metaData[categoryName])[0]; // only one resource in analytics
+    const resourceMeta = metaData[categoryName][r];
+    if (resourceMeta.output) {
+      amplifyConfig[categoryName] = {};
+      amplifyConfig[categoryName].plugins = {};
```

Comment:
The first parts of all ```construct``` methods are the same, checking variables, category, etc. It could be extracted into an inline function and reused.

On the other hand this line the ```plugins``` assignment is different here, from the other functions below, why?

Reply:
It's how the metadata is structured, and how they want the configuration file is structured. There is no consistent pattern cross all the categories. 
Code re-use is hard for this.